### PR TITLE
audit: whitelist bash-completion@* to use conflicts_with

### PR DESCRIPTION
**Pull request by JCount**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----
this is necessary because they install conflicting scripts into HOMEBREW_PREFIX/etc/profile.d

I changed how the whitelisted formulae are stored, both to improve readability and to make it easier to potentially add more formulae in the future if necessary. I tried to make the return conditional compact, but I am a bit uncertain if it conforms to the style of the rest of the code, so any feedback there would be welcome.

bash-completion@2 PR this is meant to address: https://github.com/Homebrew/homebrew-core/pull/10594

### CODE REVIEW

File: Library/Homebrew/dev-cmd/audit.rb [hunk 1]
```diff
@@ -532,8 +532,11 @@ def audit_conflicts
       end
     end
 
+    versioned_conflicts_whitelist = %w[node@ bash-completion@]
+
     return unless formula.conflicts.any? && formula.versioned_formula?
-    return if formula.name.start_with? "node@"
+    versioned_conflicts_whitelist.each { |c|
+      return if formula.name.start_with? c }
```
Comment (reitermarkus) [inline Library/Homebrew/dev-cmd/audit.rb]: You can do this:

```ruby
return if formula.name.start_with?(*versioned_conflicts_whitelist)
```

Comment (JCount) [inline Library/Homebrew/dev-cmd/audit.rb]: I wasn't certain if homebrew used "splat" much, so I wrote a longer version

Comment (reitermarkus) [inline Library/Homebrew/dev-cmd/audit.rb]: > I wasn't certain if homebrew used "splat" much

I think it is, but even if it wasn't, I would say in this case it is certainly more readable.

Comment (JCount) [inline Library/Homebrew/dev-cmd/audit.rb]: That is a very good point. 👍

File: Library/Homebrew/dev-cmd/audit.rb [hunk 2]
```diff
@@ -532,8 +532,11 @@ def audit_conflicts
       end
     end
 
+    versioned_conflicts_whitelist = %w[node@ bash-completion@]
```
Comment (reitermarkus) [inline Library/Homebrew/dev-cmd/audit.rb]: Maybe make this into a constant?

Comment (JCount) [inline Library/Homebrew/dev-cmd/audit.rb]: I assume you mean freezing it to make it an immutable array, not making it a true constant.

Comment (reitermarkus) [inline Library/Homebrew/dev-cmd/audit.rb]: I actually meant making it a true constant, since a whitelist seems to me like it should be a constant.

Comment (JCount) [inline Library/Homebrew/dev-cmd/audit.rb]: Ok. I can change it to that, but the issue then is that it would need to be moved out method, and into the FormulaAuditor class scope to prevent a 'dynamic constant assignment' error. Also, there are other whitelists that aren't made into constants, https://github.com/Homebrew/brew/blob/master/Library/Homebrew/dev-cmd/audit.rb#L768-L790, for example. However, it may make sense to make it a true constant. I will create a comment on the current code so that it is more visible.

File: Library/Homebrew/dev-cmd/audit.rb [hunk 3]
```diff
@@ -532,8 +532,10 @@ def audit_conflicts
       end
     end
 
+    versioned_conflicts_whitelist = %w[node@ bash-completion@].freeze
```
Comment (JCount) [inline Library/Homebrew/dev-cmd/audit.rb]: Is there any compelling reason to move this out of the `audit_conflicts` method and turn it into a true constant?

Comment (reitermarkus) [inline Library/Homebrew/dev-cmd/audit.rb]: I think it would make sense to move all of these whitelists outside, since they really are supposed to be constants, and it makes the methods easier to read if there aren't a billion lines of constants in-between “actual“ code.

Comment (JCount) [inline Library/Homebrew/dev-cmd/audit.rb]: That sounds like a perfectly valid argument to me. When I get a chance, I will transform this into a true constant. The other lists should probably be done in a separate PR I think.

Comment (reitermarkus) [inline Library/Homebrew/dev-cmd/audit.rb]: Yes, I'd leave this as is for now, and convert all of them in one go in a separate PR.
